### PR TITLE
Request Model #9

### PR DESCRIPTION
**Pull request by seblaifa3315**
### What this PR does (required):
- l created a Request model

### Any information needed to test this feature (required):
- I was not sure of the default values for the property "start" (date requested to start) and "end" (date requested to end). So, for "start" I gave a default value of current date, and  for "end" I gave a default value of current date + 1 day.

### Any issues with the current functionality (optional):
- For the properties default values of "start" and "end" I'm not sure about the implementation:

start: {
...
            default: **Date.now,**  --OR--  **Date.now(),**
        },
        end: {
...
            default: **() => Date.now() + 24x60x60x1000,**  --OR-- **Date.now() + 24x60x60x1000**
        },

### CODE REVIEW

**Comment by ezeilo-su** [comment]
Hey @seblaifa3315, kindly link your ticket. Kindly follow this [guide](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) if you need help.

File: server/models/Request.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            default: False,
+        },
+        paid : {
+            type: Boolean,
+            default: False,
+        }
+    },
+    {
+        timestamps: true
+    }
```
Comment (ezeilo-su) [inline server/models/Request.js:53]: Hey @seblaifa3315, use `camelCase` for your variable names. Use the proper linters setup in your editor to format your code.

**Review by ezeilo-su — CHANGES_REQUESTED** [review]
Good progress so far.
Happy coding!

**Comment by seblaifa3315** [comment]
- I changed to camelCase
- I added some properties

File: server/models/Request.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            type: Boolean,
+            default: false,
+        },
+        paid: {
+            type: Boolean,
+            default: false,
+        }
+    },
+    {
+        timestamps: true
```
Comment (rajivtitus) [inline server/models/Request.js:66]: 👍🏽

File: server/models/Request.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
+const mongoose = require("mongoose");
+
+const requestSchema = new mongoose.Schema(
+    {
+        ownerId: {
```
Comment (rajivtitus) [inline server/models/Request.js:5]: 👍🏽

File: server/models/Request.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            type: mongoose.Schema.Types.ObjectId,
+            required: true,
+            ref: 'User'
+        },
+        ownerName: {
```
Comment (rajivtitus) [inline server/models/Request.js:10]: Let's remove a few fields from this model as we wouldn't want to duplicate data across our db. Please see below for fields to be removed:
- ownerName
- ownerPhoto
- sitterName
- location

File: server/models/Request.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        animalType: {
+            type: String,
+            enum: ['dog', 'cat', 'other'],
+            required: true,
+        },
+        usefulInfo: {
+            type: String,
+            default: '',
+        },
+        accepted: {
```
Comment (rajivtitus) [inline server/models/Request.js:52]: Additionally, we can consolidate the `accepted/declined` fields under a single enum type object.

Example: 
```
status: {
    type: String,
    enum: ['pending', 'accepted', 'declined', 'completed']
    default: 'pending'
    lowercase: true,
    required: true,
  }
```

File: server/models/Request.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        location: {
+            type: String,
+            default: '',
+        },
+        start: {
+            type: Date,
+            required: true,
+            default: Date.now,
+        },
+        end: {
```
Comment (rajivtitus) [inline server/models/Request.js:38]: Optional: We can also add a schema validation to check if the end > start

**Review by rajivtitus — COMMENTED** [review]
Good work with this Sebastien! Just a couple of comments to address here 👍🏽

**Comment by seblaifa3315** [comment]
- I removed few field as addressed in the comments above
- I combined the accepted and declined status into one field "status"
- I  added a schema validation to check if the end > start

**Review by rajivtitus — APPROVED** [review]
LGTM, good work with the changes 👍🏽

**Pull request merged**